library-no-op/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -21,7 +21,9 @@ class ChuckerInterceptor @JvmOverloads constructor(
 
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
-        val request = chain.request()
+        val request = chain.request().newBuilder()
+            .removeHeader(Chucker.SKIP_INTERCEPTOR_HEADER_NAME)
+            .build()
```

Comment:
Sorry if I came late to the game here. Just wanted to drop my opinion on the table.

I believe the `library-no-op` variant of the interceptor should not edit the request at all. I'm afraid this is going to create a precedent.

I really like the idea of having a `Skip-Chucker-Interceptor` special header, but it should be developer's duty to make sure it's applied only on debug requests and doesn't leak. We can provide a simple interceptor that will remove Chucker headers if on `BuildConfig.DEBUG`.

Reply:
Ok. But is there any point in stripping the header at all in this case? If header is not removed in the release builds it for sure does not matter if it is removed in debug and qa builds.

Reply:
Maybe this would suggest that the stripping of header should be left to the consumers if they want to clean their requests before sending the to a server? An additional interceptor as a part of the library might be an option but I think it would have to be part of both library flavors.

Reply:
Or it could be a configuration part of the `ChuckerInterceptor` that defaults to `false`.

Reply:
Yeah I think we're having usability on one side vs security on the other. I do understand that the current implementation (that I really like btw) is the best from the developer point of view.

On the other hand this implementation leaked implementation details inside the `library-no-op`. By its nature, the `-no-op` library should do nothing at all. This is not the case anymore.

Reply:
> Maybe this would suggest that the stripping of header should be left to the consumers if they want to clean their requests before sending them to a server?

That could be an option absolutely.

> An additional interceptor as a part of the library might be an option but I think it would have to be part of both library flavors.

Unfortunately yes.

Reply:
Ok, I totally get this point and will make a follow-up PR with an additional interceptor. I just want to clarify the point about header values.